src/utils/network.ts

```diff
@@ -15,6 +15,9 @@ export const requestURL = (url: string): Promise<Data> =>
     if (options.protocol === 'https:') {
       request = https.request;
     }
+	if (settings.asksBeforeRedirect){
```

Comment:
This PR basically does nothing to the actual `webContents` navigation. You are applying those headers to a `requestURL` utility function used for things like fetching the news in the new tab page. If you want to apply those headers to `webContents` requests, please follow this API: https://www.electronjs.org/docs/api/web-request#webrequestonbeforesendheadersfilter-listener